flink-1.17-connector-bigquery/flink-connector-bigquery/src/main/java/com/google/cloud/flink/bigquery/source/reader/deserializer/AvroToRowDataConverters.java

```diff
@@ -0,0 +1,406 @@
+            case BOOLEAN: // boolean
+            case INTEGER: // int
+            case INTERVAL_YEAR_MONTH: // long
+            case BIGINT: // long
+            case INTERVAL_DAY_TIME: // long
+            case FLOAT: // float
+            case DOUBLE: // double
+                return avroObject -> avroObject;
+            case DATE:
+                return AvroToRowDataConverters::convertToDate;
+            case TIME_WITHOUT_TIME_ZONE:
+                int timePrecision = ((TimeType) type).getPrecision();
+                if (timePrecision <= millisPrecision) {
+                    return avroObject -> convertToMillisTime(avroObject, millisPrecision);
+                } else if (timePrecision <= microsPrecision) {
+                    return avroObject -> convertToMicrosTime(avroObject, microsPrecision);
+                } else {
+                    throw new UnsupportedOperationException(
+                            String.format(
+                                    "TIME type with Precision > %d is not supported!",
+                                    microsPrecision));
+                }
+            case TIMESTAMP_WITH_LOCAL_TIME_ZONE:
+                int localTsPrecision = ((LocalZonedTimestampType) type).getPrecision();
+                return createTimeDatetimeConvertor(localTsPrecision);
+            case TIMESTAMP_WITHOUT_TIME_ZONE:
+                int tsPrecision = ((TimestampType) type).getPrecision();
+                return createTimeDatetimeConvertor(tsPrecision);
+            case CHAR:
+            case VARCHAR:
+                return avroObject -> StringData.fromString(avroObject.toString());
+            case BINARY:
+            case VARBINARY:
+                return AvroToRowDataConverters::convertToBytes;
+            case DECIMAL:
+                return createDecimalConverter((DecimalType) type);
+            case ARRAY:
+                return createArrayConverter((ArrayType) type);
+            case ROW:
+                return createRowConverter((RowType) type);
+            case MAP:
+            case MULTISET:
+                return createMapConverter(type);
+            case RAW:
+            default:
+                throw new UnsupportedOperationException("Unsupported type: " + type);
+        }
+    }
+
+    private static AvroToRowDataConverter createTimeDatetimeConvertor(int precision) {
+        if (precision <= millisPrecision) {
+            return avroObject ->
+                    AvroToRowDataConverters.convertToTimestamp(avroObject, millisPrecision);
+        } else if (precision <= microsPrecision) {
+            return avroObject ->
+                    AvroToRowDataConverters.convertToTimestamp(avroObject, microsPrecision);
+        } else {
+            String invalidPrecisionError =
+                    String.format(
+                            "Obtained TIMESTAMP/DATETIME value with precision ''. "
+                                    + "Precision > %d is not supported.",
```

Comment:
"Obtained" when and where? We need to convey more context in error messages.

Reply:
Applies to other exceptions in this file.

Reply:
Fixed.